### PR TITLE
[REF] mrp: remove unnecessary condition from action_unbuild

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
This PR is created as per the discussion of this PR https://github.com/odoo/odoo/pull/155751.
ping @pfertyk @yostashiro 

@qrtl

### CODE REVIEW

File: addons/mrp/models/mrp_unbuild.py [hunk 1]
```diff
@@ -199,23 +199,24 @@ def action_unbuild(self):
...
+            if not original_move:
+                move.quantity_done = float_round(move.product_uom_qty, precision_rounding=move.product_uom.rounding)
+                continue
+            needed_quantity = move.product_uom_qty
+            moves_lines = original_move.mapped('move_line_ids')
+            if move in produce_moves and self.lot_id:
+                moves_lines = moves_lines.filtered(lambda ml: self.lot_id in ml.produce_line_ids.lot_id)  # FIXME sle: double check with arm
+            for move_line in moves_lines:
+                # Iterate over all move_lines until we unbuilded the correct quantity.
+                taken_quantity = min(needed_quantity, move_line.qty_done - qty_already_used[move_line])
```
Comment (MiquelRForgeFlow) [inline addons/mrp/models/mrp_unbuild.py:213]:
```suggestion
                taken_quantity = min(needed_quantity, move_line.quantity - qty_already_used[move_line])
```

File: addons/mrp/models/mrp_unbuild.py [hunk 2]
```diff
@@ -199,23 +199,24 @@ def action_unbuild(self):
...
-                        move_line_vals = self._prepare_move_line_vals(move, move_line, taken_quantity)
-                        self.env['stock.move.line'].create(move_line_vals)
-                        needed_quantity -= taken_quantity
-                        qty_already_used[move_line] += taken_quantity
-            else:
-                move.quantity = float_round(move.product_uom_qty, precision_rounding=move.product_uom.rounding)
+            original_move = move in produce_moves and self.mo_id.move_raw_ids or self.mo_id.move_finished_ids
+            original_move = original_move.filtered(lambda m: m.product_id == move.product_id)
+            if not original_move:
+                move.quantity_done = float_round(move.product_uom_qty, precision_rounding=move.product_uom.rounding)
```
Comment (MiquelRForgeFlow) [inline addons/mrp/models/mrp_unbuild.py:205]:
```suggestion
                move.quantity = float_round(move.product_uom_qty, precision_rounding=move.product_uom.rounding)
```

**Comment by pfertyk** [comment]
Thank you for a new PR :) I will get it reviewed and get back to you.

I can see that some tests are not passing, but they seem to be related to other tasks. Once those issues are fixed, please rebase your branch ;)

**Comment by amoyaux** [comment]
robodoo r+

**Comment by AungKoKoLin1997** [comment]
@amoyaux I rebased and CI is not failed anymore. Can you please try to merge again?

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**